app/src/main/java/net/tuxed/vpnconfigimporter/MainActivity.java

```diff
@@ -76,7 +72,9 @@ protected void onNewIntent(Intent intent) {
         }
         try {
             _connectionService.parseCallbackIntent(intent);
-            openFragment(new ConnectProfileFragment());
+            // Remove it so we dont parse it again.
+            intent.setData(null);
```

Comment:
Note: the data is set for this lifecycle, the intent used the start the activity is kept my the OS for activity re-creation from history list.